unix/modos: Rename unlink() to remove() to be consistent with other ports.

pfalcon opened the pull request:
We standardized to provide uos.remove(), as more obvious and user-friendly
name. That's what written in the docs. The Unix port implementation predates
this convention, so update it now.

Commit:
unix/modos: Rename unlink() to remove() to be consistent with other ports.

We standardized to provide uos.remove(), as more obvious and user-friendly
name. That's what written in the docs. The Unix port implementation predates
this convention, so update it now.

dpgeorge commented:
Thanks, it makes sense to rename it to the more friendly name.  But please can you update the io/open_append.py and io/open_plus.py tests to use rename instead of unlink so they still run on the unix port.

Commit:
tests/io: Update tests for uos.remove() instead of uos.unlink().

After Unix port switches from one to another, to be consistent with
baremetal ports.

pfalcon commented:
> But please can you update the io/open_append.py and io/open_plus.py tests to use rename

Updated.

pfalcon commented:
This appears to be merged in https://github.com/micropython/micropython/commit/80aca27a40a76e57d4ca12965d1a6f5c055d268c / https://github.com/micropython/micropython/commit/5c34c2ff7f0c44ec9e1d77059162584c6bd99c92, thanks.
